nix: don't strip if we are building a debug version of the package

jcollie opened the pull request.

qwerasd205 commented:
The condition is backwards, ***dont***Strip should be *true* if we *are* building `Debug`

jcollie committed:
nix: keep symbols if we're building a debug package

also add CI tests to make sure debug symbols exist

Co-authored-by: Mitchell Hashimoto <m@mitchellh.com>

jcollie commented:
> The condition is backwards, _**dont**_Strip should be _true_ if we _are_ building `Debug`

Doh! Fixed. Also added some CI checks to make sure that debug symbols actually make it into the package.
